Tasks/AzureVmssDeployment/operations/VirtualMachineScaleSet.ts

```diff
@@ -24,135 +25,143 @@ export default class VirtualMachineScaleSet {
-                            properties: {
-                                type: extensionMetadata.type,
-                                publisher: extensionMetadata.publisher,
-                                typeHandlerVersion: extensionMetadata.typeHandlerVersion,
-                                autoUpgradeMinorVersion: true,
-                                settings: {
-                                    "fileUris": [ this.taskParameters.customScriptUrl ]
-                                },
-                                protectedSettings: {
-                                    "commandToExecute": this.taskParameters.customScriptCommand
+                        var extensionMetadata: azureModel.VMExtensionMetadata = null;
+                        var customScriptExtension: azureModel.VMExtension = null;
+                        if(!!this.taskParameters.customScriptUrl) {
+                            extensionMetadata = this._getCustomScriptExtensionMetadata(osType);
+                            customScriptExtension = {
+                                name: "CustomScriptExtension" + Date.now().toString(),
+                                properties: {
+                                    type: extensionMetadata.type,
+                                    publisher: extensionMetadata.publisher,
+                                    typeHandlerVersion: extensionMetadata.typeHandlerVersion,
+                                    autoUpgradeMinorVersion: true,
+                                    settings: {
+                                        "fileUris": [ this.taskParameters.customScriptUrl ]
+                                    },
+                                    protectedSettings: {
+                                        "commandToExecute": this.taskParameters.customScriptCommand
+                                    }
                                 }
-                            }
-                        };
+                            };
 
-                        this._getExistingCustomScriptExtension(client, resourceGroupName, customScriptExtension, (error, matchingExtension, request, response) => {
+                            var matchingExtension = await this._getExistingCustomScriptExtension(client, resourceGroupName, customScriptExtension);
                             // if extension already exists, remove it
                             if(!!matchingExtension) {
-                                this._deleteAndInstallCustomScriptExtension(client, resourceGroupName, matchingExtension, customScriptExtension, resolve, reject, (error, matchingExtension, request, response) => {
-                                    this._updateImageInternal(client, resourceGroupName, resolve, reject);
-                                });
+                                await this._deleteAndInstallCustomScriptExtension(client, resourceGroupName, matchingExtension, customScriptExtension);
```

Comment:
P2: similarly consider if you need to merge this method now. Or you can delete if a matching extension is found. Install extension always. And then update image.